x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/SecurityIndexReaderWrapper.java

```diff
@@ -50,20 +52,22 @@
     private final XPackLicenseState licenseState;
     private final SecurityContext securityContext;
     private final ScriptService scriptService;
+    private final Settings settings;
 
     public SecurityIndexReaderWrapper(Function<ShardId, SearchExecutionContext> searchExecutionContextProvider,
                                       DocumentSubsetBitsetCache bitsetCache, SecurityContext securityContext,
-                                      XPackLicenseState licenseState, ScriptService scriptService) {
+                                      XPackLicenseState licenseState, ScriptService scriptService, Settings settings) {
         this.scriptService = scriptService;
         this.searchExecutionContextProvider = searchExecutionContextProvider;
         this.bitsetCache = bitsetCache;
         this.securityContext = securityContext;
         this.licenseState = licenseState;
+        this.settings = settings;
     }
 
     @Override
     public DirectoryReader apply(final DirectoryReader reader) {
-        if (licenseState.isSecurityEnabled() == false ||
+        if (XPackSettings.SECURITY_ENABLED.get(settings) == false ||
```

Comment:
This is not necessary because the `SecurityIndexReaderWrapper` is not instantiated when Security is false.
Ideally there should be a test for it in `xpack.security.SecurityTests` .

Reply:
> there should be a test for it 

"it" being what in this context?

Reply:
Sorry for not being clear.
I mean that the `Security` plugin's `onIndexModule` method doesn't do anything if security is disabled.

Reply:
Added a test in 1c98a3c, let me know if you can think of a better way to test this